cmd/contour/serve.go

```diff
@@ -346,14 +349,10 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 	g.Add(func(stop <-chan struct{}) error {
 		log := log.WithField("context", "grpc")
 
-		synced := make([]cache.InformerSynced, 0, len(informers))
-		for _, inf := range informers {
-			synced = append(synced, inf.HasSynced)
-		}
-
 		log.Printf("waiting for informer caches to sync")
-		if !cache.WaitForCacheSync(stop, synced...) {
-			return fmt.Errorf("error waiting for cache to sync")
+		err := synclist.WaitForSync(stop)
```

Comment:
might be easier to read if you use the 

if err := ... ; err != nil {
return err
}

form.

Also, check(err) should be ok to use here.

Reply:
I used `return err` because the rest of the function did, seems weird to use check inside an inner function to me.